internal/kuberesource/mutators.go

```diff
@@ -195,27 +231,33 @@ func PatchServiceMeshAdminInterface(resources []any, port int32) []any {
 }
 
 // MapPodSpec applies a function to a PodSpec in a Kubernetes resource.
-func MapPodSpec(resource any, f func(spec *applycorev1.PodSpecApplyConfiguration) *applycorev1.PodSpecApplyConfiguration) any {
+func MapPodSpec(
```

Comment:
Have you considered adding a `MapPodSpecWithMeta` and implementing the existing `MapPodSpec` with it?